node/manager.go

```diff
@@ -0,0 +1,149 @@
+	"github.com/iotexproject/iotex-core/pkg/log"
+	"github.com/iotexproject/iotex-core/pkg/routine"
+	"github.com/iotexproject/iotex-core/pkg/util/byteutil"
+	"github.com/iotexproject/iotex-core/pkg/version"
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type (
+	transmitter interface {
+		BroadcastOutbound(context.Context, proto.Message) error
+		UnicastOutbound(context.Context, peer.AddrInfo, proto.Message) error
+	}
+
+	heightable interface {
+		TipHeight() uint64
+	}
+
+	privateKey interface {
+		PublicKey() crypto.PublicKey
+		Sign([]byte) ([]byte, error)
+	}
+
+	// DelegateManager manage delegate node info
+	DelegateManager struct {
+		cfg         Config
+		nodeMap     map[string]iotextypes.NodeInfo
+		broadcaster *routine.RecurringTask
+		transmitter transmitter
+		heightable  heightable
+		privKey     privateKey
+	}
+)
+
+// NewDelegateManager new delegate manager
+func NewDelegateManager(cfg *Config, t transmitter, h heightable, privKey privateKey) *DelegateManager {
+	dm := &DelegateManager{
+		cfg:         *cfg,
+		nodeMap:     make(map[string]iotextypes.NodeInfo),
+		transmitter: t,
+		heightable:  h,
+		privKey:     privKey,
+	}
+	// disable broadcast if RequestNodeInfoInterval == 0
+	if cfg.RequestNodeInfoInterval > 0 {
+		dm.broadcaster = routine.NewRecurringTask(dm.RequestNodeInfo, cfg.RequestNodeInfoInterval)
+	}
+
+	return dm
+}
+
+// Start start delegate broadcast task
+func (dm *DelegateManager) Start(ctx context.Context) error {
+	if dm.broadcaster != nil {
+		return dm.broadcaster.Start(ctx)
+	}
+	return nil
+}
+
+// Stop stop delegate broadcast task
+func (dm *DelegateManager) Stop(ctx context.Context) error {
+	if dm.broadcaster != nil {
+		return dm.broadcaster.Stop(ctx)
+	}
+	return nil
+}
+
+// HandleNodeInfo handle node info message
+func (dm *DelegateManager) HandleNodeInfo(ctx context.Context, peerID string, msg *iotextypes.ResponseNodeInfoMessage) {
+	// verify signature
+	pubKey, err := crypto.BytesToPublicKey(msg.Pubkey)
+	if err != nil {
+		log.L().Warn("delegate manager convert to publick key failed", zap.Error(err))
+		return
+	}
+	hash := hashNodeInfo(msg)
+	if !pubKey.Verify(hash[:], msg.Signature) {
+		log.L().Warn("delegate manager node info message verify failed", zap.String("peer", peerID))
+		return
+	}
+
+	dm.UpdateNode(pubKey.Address().String(), msg.Info)
+}
+
+// UpdateNode update node info
+func (dm *DelegateManager) UpdateNode(addr string, node *iotextypes.NodeInfo) {
+	// update dm.nodeMap
+	dm.nodeMap[addr] = *node
+	// update metric
+	nodeDelegateHeightGauge.WithLabelValues(addr, node.Version).Set(float64(node.Height))
+}
+
+// RequestNodeInfo broadcast request node info message
+func (dm *DelegateManager) RequestNodeInfo() {
+	log.L().Info("delegate manager request node info")
+
+	// broadcast request meesage
```

Comment:
The request message should be unicasted to the target, not broadcasted to the network.

Reply:
You may have some misunderstanding, the general mechanism is as follows:
- Firstly those nodes who are configured will broadcast request message periodically
- And then, the node who recievied request message will unicast node info to the request sender

Reply:
As separate discussion, broadcast mechanism revised to node broadcast node info msg  proactively and periodically